### PR TITLE
MNT: Updating to use pymsis wheels instead of manual layers

### PR DESCRIPTION
**Pull request by greglucas**
This enables pymsis to be added to the requirements.txt file and
get installed and uploaded as a bundle. Before we had to manually
create the zip layer and compile the binaries to attach properly.

To test it:
```bash
sam build
sam local invoke MSIS2Function --event events/event_msis.json
sam local start-api
# Then use a GET request (I use postman) to:
# http://127.0.0.1:3000/msis2/surface?date=2018-01-01T12:00&altitude=400&f107=146.7&f107a=163.6666&ap=7&options=[0,0,0,0,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1]
```

### CODE REVIEW

**Comment by greglucas** [comment]
@bmcclellan-cu I'm pinging you to try this out locally and check out the previous SAM work I did.

I'd also like to put this over into our dev - staging - prod pipeline in the new account as well. So, if you have ideas for how best to move this over too, that'd be great!

**Comment by greglucas** [comment]
@bmcclellan-cu I was trying to get this to work with our dev account and having a nice domain name. I was able to get it working with a cross-stack reference for the signing certificate. So now, the URL looks like:

https://msis-api.dev.swx-trec.com/msis2/altitude?date=2018-01-01T12:00&longitude=0&latitude=0&f107=146.7&f107a=163.6666&ap=7&options=[0,0,0,0,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1]

Which the frontend API can now point to as a static version instead of the cryptic default version that can get updated on new deploys too.

File: template.yaml [hunk 1]
```diff
@@ -11,46 +11,49 @@ Globals:
     Timeout: 3
 
 Resources:
+  MSIS2Api:
+    Type: AWS::Serverless::Api
+    Properties:
+      StageName: dev
+      Domain:
+        DomainName: msis-api.dev.swx-trec.com
+        CertificateArn: !ImportValue accountCertificateArn
```
Comment (greglucas) [inline template.yaml:20]: This is what I found nice and didn't realize we could do with the outputs from other cloud formation stacks. I thought we had to hard-code them in after the output, but they are actually stored out in AWS for other stacks to reference and use.

**Comment by bmcclellan-cu** [comment]
Hi @greglucas, this was on my to-do list but somehow got skipped, I'm getting up to speed on SAM now.

The first thing I notice is that the sam build will fail, if your docker instance isn't authenticated against the public AWS ECR. This fixes that, but the error message is not clear and google doesn't provide any clear solutions. Digging through the errors you do find:  ("pull access denied for public.ecr.aws/sam/build-python3.8, repository does not exist or may require 'docker login': denied: Your authorization token has expired. Reauthenticate and try again.")

 
aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin public.ecr.aws

**Comment by greglucas** [comment]
That's interesting, I didn't realize you'd need to be authenticated for anything here unless you wanted to deploy... The 3.8 error is odd too because I updated this to 3.9, so were you on main branch or this one? Might need to do a 'pip install aws-sam-cli --upgrade' too in a new environment for this project. Happy to hop on a call sometime this next week to see what the errors you're getting are.

**Review by bmcclellan-cu — APPROVED** [review]
Looks good! I was originally getting setup with master, but switched to this branch and it works.

(The docker image it pulls during the initial SAM build is to simulate the Lambda environment, so it's needed when the build command is run).

Testing Lamdba's via the cdk utilize these same docker containers to simulate Lambda environments (it looks like you can flag if you want to test against the cloud or have all testing be 100% local).

**Pull request merged**